[jit] remove Dict iterationOrder and use insertion order

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36696 [jit] Add dictionary as output of tracer
* **#36609 [jit] remove Dict iterationOrder and use insertion order**



This PR remove iterationOrder that backed by CompareKeys, we universally
use Dict inseration order default backed by c10::Dict to match the
python behaviors

Differential Revision: [D21056963](https://our.internmc.facebook.com/intern/diff/D21056963)

wanchaol committed:
[jit] add comment for Dict iterationOrder

[ghstack-poisoned]

jamesr66a reviewed (dismissed):
Great!

wanchaol committed:
Update on "[jit] add comment for Dict iterationOrder"




[ghstack-poisoned]

wanchaol committed:
Update on "[jit] remove Dict iterationOrder and use insertion order"




This PR remove iterationOrder that backed by CompareKeys, we universally
use Dict inseration order default backed by c10::Dict to match the
python behaviors

[ghstack-poisoned]

jamesr66a reviewed (approved):
Seems fine, just make sure to watch out for any incompatibility that might arise in existing models. It's pretty unlikely anyone's relying on the `iterationOrder` ordering, but it's worth watching out for

wanchaol committed:
Update on "[jit] remove Dict iterationOrder and use insertion order"




This PR remove iterationOrder that backed by CompareKeys, we universally
use Dict inseration order default backed by c10::Dict to match the
python behaviors

Differential Revision: [D21056963](https://our.internmc.facebook.com/intern/diff/D21056963)

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] remove Dict iterationOrder and use insertion order"




This PR remove iterationOrder that backed by CompareKeys, we universally
use Dict inseration order default backed by c10::Dict to match the
python behaviors

Differential Revision: [D21056963](https://our.internmc.facebook.com/intern/diff/D21056963)

[ghstack-poisoned]
